src/coord/src/catalog/builtin.rs

```diff
@@ -0,0 +1,465 @@
+};
+
+pub const MZ_SCHEDULING_PARKS: BuiltinLog = BuiltinLog {
+    name: "mz_scheduling_parks",
+    variant: LogVariant::Timely(TimelyLog::Parks),
+    id: GlobalId::System(1010),
+    index_id: GlobalId::System(1011),
+};
+
+pub const MZ_ARRANGEMENT_SIZES: BuiltinLog = BuiltinLog {
+    name: "mz_arrangement_sizes",
+    variant: LogVariant::Differential(DifferentialLog::Arrangement),
+    id: GlobalId::System(1012),
+    index_id: GlobalId::System(1013),
+};
+
+pub const MZ_ARRANGEMENT_SHARING: BuiltinLog = BuiltinLog {
+    name: "mz_arrangement_sharing",
+    variant: LogVariant::Differential(DifferentialLog::Sharing),
+    id: GlobalId::System(1014),
+    index_id: GlobalId::System(1015),
+};
+
+pub const MZ_MATERIALIZATIONS: BuiltinLog = BuiltinLog {
+    name: "mz_materializations",
+    variant: LogVariant::Materialized(MaterializedLog::DataflowCurrent),
+    id: GlobalId::System(1016),
+    index_id: GlobalId::System(1017),
+};
+
+pub const MZ_MATERIALIZATION_DEPENDENCIES: BuiltinLog = BuiltinLog {
+    name: "mz_materialization_dependencies",
+    variant: LogVariant::Materialized(MaterializedLog::DataflowDependency),
+    id: GlobalId::System(1018),
+    index_id: GlobalId::System(1019),
+};
+
+pub const MZ_WORKER_MATERIALIZATION_FRONTIERS: BuiltinLog = BuiltinLog {
+    name: "mz_worker_materialization_frontiers",
+    variant: LogVariant::Materialized(MaterializedLog::FrontierCurrent),
+    id: GlobalId::System(1020),
+    index_id: GlobalId::System(1021),
+};
+
+pub const MZ_PEEK_ACTIVE: BuiltinLog = BuiltinLog {
+    name: "mz_peek_active",
+    variant: LogVariant::Materialized(MaterializedLog::PeekCurrent),
+    id: GlobalId::System(1022),
+    index_id: GlobalId::System(1023),
+};
+
+pub const MZ_PEEK_DURATIONS: BuiltinLog = BuiltinLog {
+    name: "mz_peek_durations",
+    variant: LogVariant::Materialized(MaterializedLog::PeekDuration),
+    id: GlobalId::System(1024),
+    index_id: GlobalId::System(1025),
+};
+
+lazy_static! {
+    pub static ref MZ_VIEW_KEYS: BuiltinTable = BuiltinTable {
+        name: "mz_view_keys",
+        desc: RelationDesc::empty()
+            .with_column("global_id", ScalarType::String.nullable(false))
+            .with_column("column", ScalarType::Int64.nullable(false))
+            .with_column("key_group", ScalarType::Int64.nullable(false)),
+        id: GlobalId::System(2001),
+        index_id: GlobalId::System(2002),
+    };
+    pub static ref MZ_VIEW_FOREIGN_KEYS: BuiltinTable = BuiltinTable {
+        name: "mz_view_foreign_keys",
+        desc: RelationDesc::empty()
+            .with_column("child_id", ScalarType::String.nullable(false))
+            .with_column("child_column", ScalarType::Int64.nullable(false))
+            .with_column("parent_id", ScalarType::String.nullable(false))
+            .with_column("parent_column", ScalarType::Int64.nullable(false))
+            .with_column("key_group", ScalarType::Int64.nullable(false))
+            .with_key(vec![0, 1, 4]),
```

Comment:
I know this was this way previously but this key definition jumps out as unclear - maybe a comment here about why (child_id, child_column, key_group) constitute a key would be nice? we've definitely messed this up in the past

Reply:
I'm... afraid I have no idea. I left a TODO.